pkg/provider/electricity_maps.go

```diff
@@ -64,34 +64,139 @@ func (e *ElectricityMapsClient) GetCarbonIntensity(ctx context.Context, location
 		return nil, errBadStatus(resp)
 	}
 
-	data := &electricityMapsData{}
-	err = json.NewDecoder(resp.Body).Decode(data)
+	type EMHistoryResponse struct {
```

Comment:
```suggestion
	type electricityMapsHistoryResponse struct {
```

Rather than have this struct inline could you rename it and add it to the end of the file?

For now we have all the providers in a single package but we use the provider name as a prefix for the types.

Reply:
done!